modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -340,26 +379,55 @@ namespace {{packageName}}.Client
                     }
                 }
                 var stream = new MemoryStream(response.RawBytes);
+                log.Trace("end");
                 return stream;
             }
 
-            if (type.Name.StartsWith("System.Nullable`1[[System.DateTime")) // return a datetime object
+            if (type.Name.StartsWith("System.Nullable`1[[System.DateTime")
             {
+                // return a datetime object 
+                log.Trace("end");
                 return DateTime.Parse(response.Content,  null, System.Globalization.DateTimeStyles.RoundtripKind);
             }
 
-            if (type == typeof(String) || type.Name.StartsWith("System.Nullable")) // return primitive type
+            if (type == typeof(String) || type.Name.StartsWith("System.Nullable")
             {
+                // return primitive type 
+                log.Trace("end");
                 return ConvertType(response.Content, type);
             }
 
+            log.Trace("end");
+            return DeserializeObject(response, type);
+        }
+
+        /// <summary>
+        /// Deserialize the JSON string into a proper object.
+        /// </summary>
+        /// <typeparam name="T">Model object</typeparam>
+        /// <param name="response">The HTTP response.</param>
+        /// <returns>Object representation of the JSON string.</returns>
+        public T Deserialize<T>(IRestResponse response
+        {
+            log.Trace("start");
+            var type = typeof(T);
             // at this point, it must be a model (json)
-            try
+            if (!(DeserializeObject(response, type) is T)
             {
-                return JsonConvert.DeserializeObject(response.Content, type, serializerSettings);
+                return default(T);
```

Comment:
This is not standard behavior, and would be a breaking change.

Why not just make `DeserializeObject` generic, and perform the cast in that method? Then, you'll automatically send a 500, and you can catch the cast exception in that method if you want to expose some 4xx message.

I don't feel like we're really gaining much by changing the simple try/catch to a reusable method. Am I missing an additional usage in your PR or something?